OData/src/System.Web.OData/OData/Formatter/EdmPrimitiveHelpers.cs

```diff
@@ -84,7 +84,7 @@ public static object ConvertPrimitiveValue(object value, Type type)
                     {
                         DateTimeOffset dateTimeOffsetValue = (DateTimeOffset)value;
                         TimeZoneInfo timeZone = TimeZoneInfoHelper.TimeZone;
-                        dateTimeOffsetValue = dateTimeOffsetValue.ToUniversalTime().ToOffset(timeZone.BaseUtcOffset);
+                        dateTimeOffsetValue = TimeZoneInfo.ConvertTime(dateTimeOffsetValue, timeZone);
```

Comment:
That's invalid, sorry.  The result is *required* to be -7 for values that are during daylight time.  By trying to "resolve the daylight saving problem", you're actually ignoring the reality that daylight saving time exists, and are thus *creating* a problem.

Remember, "time zone != offset".  Setting "Pacific Time" does not set "a -8 time zone". It sets a time zone whose *standard* offset is -8, and whose *daylight* offset is -7, with specific dates and times for the transitions between the two offsets, and specific history of the changes in those transitions.  (Time zones are much more complex than just a single number can offer.)